### PR TITLE
fix error adding removing tags from article mod tools

### PR DESCRIPTION
**Pull request by aitchiss**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [X] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

When I fixed some errors adding/removing tags from the mod center in https://github.com/forem/forem/pull/14566, I inadvertently introduced that same bug over on the article page mod tools 🙈 

This PR fixes it and introduces some tests to cover this path for moderation too.

## Related Tickets & Documents

Closes https://github.com/forem/forem/issues/14754

## QA Instructions, Screenshots, Recordings

From an article page:
- Click the moderator button at the bottom right
- Add a tag to the post - it should result in a snackbar being shown, and the tag appearing on the article
- Remove a different tag from the post - it should result in a snackbar being shown, and the tag disappearing from the article

From the moderator center - `/mod`
- Repeat the above steps for any article and check it works here too

### UI accessibility concerns?

N/A - there are some accessibility issues in this feature as a whole, but nothing new introduced here

## Added/updated tests?

- [X] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [X] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

### CODE REVIEW

File: app/javascript/actionsPanel/initializeActionsPanelToggle.js [hunk 1]
```diff
@@ -28,9 +28,11 @@ export function initializeActionsPanel(user, path) {
     const modContainer = document.getElementById('mod-container');
     const panelDocument = modContainer.contentDocument;
 
-    panelDocument
-      .getElementsByClassName('close-actions-panel')[0]
-      .classList.remove('hidden');
+    if (panelDocument) {
```
Comment (aitchiss) [inline app/javascript/actionsPanel/initializeActionsPanelToggle.js:31]: This was causing a mostly harmless error when not present, but Cypress took exception to it 😅

File: app/javascript/packs/articleModerationTools.js [hunk 2]
```diff
@@ -1,28 +1,30 @@
 /* global userData */
 import { isModerationPage } from '@utilities/moderation';
 
-const user = userData();
-const { authorId: articleAuthorId, path } = document.getElementById(
-  'article-show-container',
-).dataset;
+getCsrfToken().then(() => {
```
Comment (aitchiss) [inline app/javascript/packs/articleModerationTools.js:4]: Added to make sure userdata is always available by the time we start this process

File: app/javascript/actionsPanel/actionsPanel.js [hunk 3]
```diff
@@ -203,6 +202,15 @@ function renderTagOnArticle(tagName, colors) {
   articleTagsContainer.appendChild(newTag);
 }
 
+function getArticleContainer() {
+  const articleIframe =
+    window.parent.document?.getElementsByClassName('article-iframe')[0];
```
[the remark this answers is not in the document]
Comment (aitchiss) [reply app/javascript/actionsPanel/actionsPanel.js:207]: the problem with `window.top` is that when you run in cypress, `window.top` is the cypress test runner iframe. That's why I changed it to `parent` in https://github.com/forem/forem/pull/14566 to allow us to test in Cypress 🙁

[the remark this answers is not in the document]
Comment (nickytonline) [reply app/javascript/actionsPanel/actionsPanel.js:207]: Ahh, good to know. I didn't have the full context. In that case, disregard.

**Review by nickytonline — APPROVED** [review]
LGTM and as mentioned, disregard my nitpick based on the Cypress context.

**Review by citizen428 — APPROVED** [review]
Thanks for the quick fix @aitchiss! 🚀

**Pull request merged**